### DISCUSSION TITLE
[Chapter 03]

### DISCUSSION THREAD

**Discussion by coalong** [cowaine/is-my-code-that-strange#21] [Chapter 01-03 (설계)]
Q1.  페이지 22쪽을 보면 아래와 같은 내용이 있습니다.

> <<객체 지향 소프트웨어 설계 2판 원칙과 개념>>에서는 "클래스, 단언(assertion), 제네릭, 상속, 다형성, 동적 바인딩으로 구성되는 것"  이라고 나와있습니다.

여기서 단언(assertion)이란 무엇이며, 객체지향설계에 이 요소를 왜 언급했을까요?

Q2. 페이지 26쪽에 Money 클래스를 정의하고 인스턴스 변수로 amount(금액)을 int 자료형으로 갖고 있습니다.

금액의 타입이 int 가 가장 적절한 자료형일지 궁금합니다.

**Comment by Irisation23** [comment]
Q2. https://stackoverflow.com/questions/5356123/why-dont-applications-typically-use-int-to-internally-represent-currency-values

**Reply by Irisation23** [reply]
Q2. 

```javascript
0.3 + 0.3 + 0.3
// 0.8999999999999 나옴 (부동소수점 Issue)
```

**Reply by Irisation23** [reply]
Q2.

varchar로 금액을 저장해서 이슈가 생길 수 있음

**Comment by Irisation23** [comment]
Q1. https://stackoverflow.com/questions/2758224/what-are-assertions-in-java-and-when-should-they-be-used
